app/validation/validator.py

```diff
@@ -1,6 +1,98 @@
+from app.validation.validation_result import ValidationResult
+from app.validation.required import Required
+from app.validation.type_validator_factory import TypeValidatorFactory
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+class ValidationException(Exception):
+    pass
+
+
```

Comment:
I wonder if this "component" should be called ValidationEngine rather than Validator, then the file renamed validation.py and the i_validator.py renamed to validator.py. Ties in with the RoutingEngine naming convention then. What do you think?